src/valid/analyzer.rs

```diff
@@ -140,7 +140,7 @@ pub struct SamplingKey {
 pub struct ExpressionInfo {
     pub uniformity: Uniformity,
     pub ref_count: usize,
-    assignable_global: Option<Handle<crate::GlobalVariable>>,
+    pub assignable_global: Option<Handle<crate::GlobalVariable>>,
```

Comment:
@kvark It's ok to make it public?

Reply:
I don't think there is a strong case here for exposing it. You could just match the expression itself, instead.